server/src/main/java/io/unitycatalog/server/persist/dao/IdentifiableDAO.java

```diff
@@ -0,0 +1,42 @@
+package io.unitycatalog.server.persist.dao;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import jakarta.persistence.Column;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.SuperBuilder;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@MappedSuperclass
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@SuperBuilder
+public class IdentifiableDAO {
+    @Id
+    @Column(name = "id")
+    private UUID id;
+
+    @Column(name = "name", nullable = false)
+    private String name;
+
+    public static <T extends IdentifiableDAO> Optional<String> getParentIdColumnName(Class<T> entityClass) {
+        if (TableInfoDAO.class == entityClass
```

Comment:
We should just return `empty()` here and let each DAO class override the function if needed. A parent class usually should not refer to subclasses.

Reply:
The issue with this is that we never actually have an object of the entityClass in `PagedListingHelper`. So the way I can do this is using reflection by creating a dummy object and then calling the overloaded function using that dummy object. let me know if that approach is fine.

Reply:
Ah, didn't realize this is method is static. Probably that's the best we can do now. Later we might want to move this to an "EntityDescriptor" instance.